flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java

```diff
@@ -106,8 +106,16 @@ public void testDetachedMode() throws InterruptedException, IOException {
 			}
 		}
 
-		//additional sleep for the JM/TM to start and establish connection
-		sleep(3000);
+		// additional sleep for the JM/TM to start and establish connection
+		long startTime = System.currentTimeMillis();
+		while (System.currentTimeMillis() - startTime < 10000 &&
```

Comment:
Let's use `System.nanoTime` for the deadline. `nanoTime` is not susceptible to clock shifts.

Reply:
done